address_translators.go

```diff
@@ -0,0 +1,53 @@
+package gocql
+
+import (
+	"log"
+	"net"
+)
+
+// AddressTranslator provides a way translate node addresses (and ports) when one is
+// discovered or received as a node event. This is especially useful in ec2 (when
+// using the EC2MultiRegionAddressTranslator) to translate public IPs to private IPs
+// when possible.
+type AddressTranslator interface {
+	// Translate will translate the provided address and/or port when a host is discovered.
+	Translate(addr string, port int) (string, int)
```

Comment:
The addr here should always be a `net.IP` actually, so I think the function should look something like

```Translate(ip net.IP, port int) (ip net.IP, port int, err error)`

Returning an error will be useful as it will prevent the driver having to check down the chain if an address is valid and such, also allowing the translate the block hosts (though arguably that should be done by the host filter).